shopinvader/models/shopinvader_variant.py

```diff
@@ -278,4 +278,19 @@ def get_shop_data(self):
     def _get_shop_data(self):
         """Compute shop data base_jsonify parser."""
         exporter = self.env.ref("shopinvader.ir_exp_shopinvader_variant")
-        return self.jsonify(exporter.get_json_parser(), one=True)
+
+        # Same logic as the search-engine implementation
```

Comment:
this is wrong IMO. It should stay in the parser.
It should default to `id` in the search engine implementation IMO.

Reply:
Indeed it's weird to have this hack here.
So the solution is to remove the magic addition of the mapping id in the search engine here : https://github.com/OCA/search-engine/pull/96/files#diff-911701f00b818f1c2893a3a3fd17ccb4b26dfe13a6f7126f6741e1887a7e7691R28
And then in the search engine raise an error if the key "id" is missing, ok for you ?

Reply:
yes

Reply:
I have a better solution
I can just define a export line for the product id

Reply:
done !